File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.java
class Solution {
    public int minOperations(int[] nums, int x) {
        int total = 0;
        int n = nums.length;
        for(int num : nums)
            total+= num;
        int target = total - x;
        if (target < 0) return -1;
        if (target == 0) return n;
        int max_sum = -1;
        int sum = 0;
        int left = 0;
        for(int right = 0; right<n; right++){
            sum+= nums[right];
            while(left<=right && sum > target){
                sum -= nums[left];
                left++;
            }
            if (sum == target){
                max_sum = Math.max(max_sum, right - left + 1);
            }
               
        }
        if (max_sum == -1) 
                return -1;
        else
            return n - max_sum;     
    }
}

File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.py
class Solution:
    def minOperations(self, nums: list[int], x: int) -> int:
        n = len(nums)
        total = 0
        for num in nums:
            total+=num
        target = total - x
        if target < 0:
            return -1
        if target == 0:
            return n
        left = 0
        curr_sum = 0
        max_sum = -1
        for right in range(n):
            curr_sum += nums[right]
            while left <= right and curr_sum > target:
                curr_sum -= nums[left]
                left += 1
            if curr_sum == target:
                max_sum = max(max_sum, right - left + 1)   
        if max_sum == -1:
            return -1
        else:
            return n - max_sum                      